packages/core/src/metrics/MetricsService.ts

```diff
@@ -19,18 +21,19 @@ export class MetricsService {
 
   protected publisher?: MetricsPublisher;
   protected configuration?: ServiceConfiguration;
-  private readonly defaultMetrics: Metrics[];
-
-  constructor(appConfig: AeroGearConfiguration) {
-    const configuration = new ConfigurationHelper(appConfig).getConfigByType(MetricsService.TYPE);
-    this.defaultMetrics = this.buildDefaultMetrics();
+  private readonly defaultMetrics?: Metrics[];
```

Comment:
I thought the consensus was to have defaultMetrics no matte what?

Reply:
IF metrics are disabled then makes no sense. However we can check if they are enabled before creating object. Then logic will be much simpler and we will avoid optionals.

Reply:
@wtrocki I thought if metrics was disable (i.e. no config present), metrics would be logged in the console anyway. But don't get me wrong, it is OK for me having this as optional, I was just curious.

Reply:
> metrics would be logged in the console anyway.